### PR TITLE
Add RequestConvertible and Tests

### PR DESCRIPTION
**Pull request by dcvz**
Issue: #3

### CODE REVIEW

File: SoundCloudSwift/Source/Core/Entities/SCSRequestConvertible.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//
+//  Created by David Chavez on 9/29/15.
+//  Copyright © 2015 SugarTeam. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+/// Request creation struct
+struct SCSRequestConvertible {
```
Comment (pepibumur) [inline SoundCloudSwift/Source/Core/Entities/SCSRequestConvertible.swift]: I think this struct should be private, the Framework is not going to expose the way the requests are built.

**Comment by pepibumur** [comment]
Nice job @dcvz

File: SoundCloudSwift/Source/Core/Entities/SCSRequestConvertible.swift (hunk 1 above)
Comment (dcvz) [inline SoundCloudSwift/Source/Core/Entities/SCSRequestConvertible.swift]: It's internal by default so won't be exposed outside the framework. Making it private would require to be in the same file as that using it @pepibumur

Comment (pepibumur) [inline SoundCloudSwift/Source/Core/Entities/SCSRequestConvertible.swift]: Oh yes sorry, you're right. 
What do you think about not using Prefixes anymore @dcvz ?

Comment (dcvz) [inline SoundCloudSwift/Source/Core/Entities/SCSRequestConvertible.swift]: Maybe for internals we don't need them, but don't you think it'd be more convenient for public classes? For example if we had a ```User``` class and they had one, it'd be annoying to have to specify ```SoundCloudSwift.User```. What are your thoughts @pepibumur? I think I'd be ok with removing the prefixes.

**Pull request closed without merging**